In [1]:
import pandas as pd
# Librerías usadas para estandarizar la información
import repositories.dicts as dicts
import repositories.utils as utils
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np


# Declaring the names of the columns we're interested in

In [2]:
columnas=['DP','DPNOM','DPMP','MPIO','AÑO','ÁREA GEOGRÁFICA']+[str(5*i)+"_"+str(5*(i+1)) for i in range(20)]+["Mujeres_100"]


# Reading the population by genre an municipalities for the period considered

df_1995_2004=pd.DataFrame(pd.read_excel("anexo-area-sexo-edad-proyecciones-poblacion-Municipal_1995-2004.xlsx"))
df_2004_2017=pd.DataFrame(pd.read_excel("anexo-area-sexo-edad-proyecciones-poblacion-Municipal_2005-2017.xlsx"))
df_2018=pd.DataFrame(pd.read_excel("anexo-proyecciones-poblacion-Municipal_2018-2026.xlsx"))

In [3]:
df_1995_2004=pd.DataFrame(pd.read_csv("data/anexo-area-sexo-edad-proyecciones-poblacion-Municipal_1995-2004.csv"))
df_2004_2017=pd.DataFrame(pd.read_csv("data/anexo-area-sexo-edad-proyecciones-poblacion-Municipal_2005-2017.csv"))
df_2018=pd.DataFrame(pd.read_csv("data/anexo-proyecciones-poblacion-Municipal_2018-2026.csv"))

df_1995_2004.to_csv("anexo-area-sexo-edad-proyecciones-poblacion-Municipal_1995-2004.csv")
df_2004_2017.to_csv("anexo-area-sexo-edad-proyecciones-poblacion-Municipal_2005-2017.csv")
df_2018.to_csv("anexo-proyecciones-poblacion-Municipal_2018-2026.csv")



# Standardizing the names and codes of the municipalities, filtering data associated to the whole municipalities.

In [4]:
df_1995_2004=df_1995_2004[df_1995_2004['ÁREA GEOGRÁFICA']=="Total"].copy()
df_1995_2004.DPMP=df_1995_2004.DPMP.apply(lambda x: str(x).zfill(5))

In [5]:
df_2004_2017=df_2004_2017[df_2004_2017['ÁREA GEOGRÁFICA']=="Total"].copy()
df_2004_2017.DPMP=df_2004_2017.DPMP.apply(lambda x: str(x).zfill(5))

In [6]:
df_2018=df_2018[df_2018['ÁREA GEOGRÁFICA']=="Total"].copy()
df_2018.DPMP=df_2018.DPMP.apply(lambda x: str(x).zfill(5))


# Selecting data associated to woman populations during the time horizon considered

In [7]:
for i in range(20): 
    df_1995_2004[str(5*i)+"_"+str(5*(i+1))]=sum(df_1995_2004["Mujeres_"+str(j)] for j in range(i*5,(i+1)*5))


In [8]:
for i in range(20): 
    df_2004_2017[str(5*i)+"_"+str(5*(i+1))]=sum(df_2004_2017["Mujeres_"+str(j)] for j in range(i*5,(i+1)*5))

In [9]:
for i in range(20): 
    df_2018[str(5*i)+"_"+str(5*(i+1))]=sum(df_2018["Mujeres_"+str(j)] for j in range(i*5,(i+1)*5))

# Selecting the subset of columns we are interested in

In [10]:
df_1995_2004=df_1995_2004[columnas]
df_2004_2017=df_2004_2017[columnas]
df_2018=df_2018[columnas]

# Concatenating dataframes and saving it to an external file for further processing (if necessary)

In [11]:
df=pd.concat([df_1995_2004,df_2004_2017,df_2018])


In [12]:

df["yes"]=df["DPMP"].astype(int).isin(dicts.municipios_segregantes_unique.keys())
df["oldcode"]=df["DPMP"].copy()
df["DPMP"]=df["oldcode"].apply(lambda x: str(dicts.municipios_segregantes_unique[int(x)]).zfill(5) if (int(x) in dicts.municipios_segregantes_unique.keys()) else x)

df["DPMP"]=df["DPMP"].apply(lambda x: str(int(x)).zfill(5))
#df["DPMP"],df["oldcode"]=df["oldcode"].copy(),df["DPMP"].copy()


display(df[df["yes"]])
df.to_csv("población_por_grupo.csv")
df[df.DPMP!=df.oldcode]



DP    DPNOM   DPMP           MPIO   AÑO ÁREA GEOGRÁFICA  0_5  5_10  \
1720   13  Bolívar  13600         Norosí  1995           Total    0     0   
1721   13  Bolívar  13600         Norosí  1996           Total    0     0   
1722   13  Bolívar  13600         Norosí  1997           Total    0     0   
1723   13  Bolívar  13600         Norosí  1998           Total    0     0   
1724   13  Bolívar  13600         Norosí  1999           Total    0     0   
...    ..      ...    ...            ...   ...             ...  ...   ...   
10057  97   Vaupés  97001  Yavaraté (CD)  2022           Total   57   111   
10058  97   Vaupés  97001  Yavaraté (CD)  2023           Total   61   113   
10059  97   Vaupés  97001  Yavaraté (CD)  2024           Total   60   113   
10060  97   Vaupés  97001  Yavaraté (CD)  2025           Total   60   116   
10061  97   Vaupés  97001  Yavaraté (CD)  2026           Total   58   119   

       10_15  15_20  ...  65_70  70_75  75_80  80_85  85_90  90_95  95_100  \
1720       0      0  ...      0      0      0      0      0      0       0   
1721       0      0  ...      0      0      0      0      0      0       0   
1722       0      0  ...      0      0      0      0      0      0       0   
1723       0      0  ...      0      0      0      0      0      0       0   
1724       0      0  ...      0      0      0      0      0      0       0   
...      ...    ...  ...    ...    ...    ...    ...    ...    ...     ...   
10057    129    104  ...      4      4      7     12      3      4       2   
10058    131    111  ...      5      4      7     12      3      4       3   
10059    141    114  ...     13     11      8      5      5      1       0   
10060    150    124  ...     15     11      9      5      5      1       0   
10061    158    132  ...     15     12      9      5      5      0       0   

       Mujeres_100   yes  oldcode  
1720             0  True    13490  
1721             0  True    13490  
1722             0  True    13490  
1723             0  True    13490  
1724             0  True    13490  
...            ...   ...      ...  
10057            0  True    97889  
10058            0  True    97889  
10059            0  True    97889  
10060            0  True    97889  
10061            0  True    97889  

[1856 rows x 29 columns]

DP    DPNOM   DPMP           MPIO   AÑO ÁREA GEOGRÁFICA  0_5  5_10  \
1720   13  Bolívar  13600         Norosí  1995           Total    0     0   
1721   13  Bolívar  13600         Norosí  1996           Total    0     0   
1722   13  Bolívar  13600         Norosí  1997           Total    0     0   
1723   13  Bolívar  13600         Norosí  1998           Total    0     0   
1724   13  Bolívar  13600         Norosí  1999           Total    0     0   
...    ..      ...    ...            ...   ...             ...  ...   ...   
10057  97   Vaupés  97001  Yavaraté (CD)  2022           Total   57   111   
10058  97   Vaupés  97001  Yavaraté (CD)  2023           Total   61   113   
10059  97   Vaupés  97001  Yavaraté (CD)  2024           Total   60   113   
10060  97   Vaupés  97001  Yavaraté (CD)  2025           Total   60   116   
10061  97   Vaupés  97001  Yavaraté (CD)  2026           Total   58   119   

       10_15  15_20  ...  65_70  70_75  75_80  80_85  85_90  90_95  95_100  \
1720       0      0  ...      0      0      0      0      0      0       0   
1721       0      0  ...      0      0      0      0      0      0       0   
1722       0      0  ...      0      0      0      0      0      0       0   
1723       0      0  ...      0      0      0      0      0      0       0   
1724       0      0  ...      0      0      0      0      0      0       0   
...      ...    ...  ...    ...    ...    ...    ...    ...    ...     ...   
10057    129    104  ...      4      4      7     12      3      4       2   
10058    131    111  ...      5      4      7     12      3      4       3   
10059    141    114  ...     13     11      8      5      5      1       0   
10060    150    124  ...     15     11      9      5      5      1       0   
10061    158    132  ...     15     12      9      5      5      0       0   

       Mujeres_100   yes  oldcode  
1720             0  True    13490  
1721             0  True    13490  
1722             0  True    13490  
1723             0  True    13490  
1724             0  True    13490  
...            ...   ...      ...  
10057            0  True    97889  
10058            0  True    97889  
10059            0  True    97889  
10060            0  True    97889  
10061            0  True    97889  

[1856 rows x 29 columns]

In [13]:
df[(df.DPMP=="27001") & (df["AÑO"]==2000)]

DP  DPNOM   DPMP          MPIO   AÑO ÁREA GEOGRÁFICA   0_5  5_10  10_15  \
5745  27  Chocó  27001        Quibdó  2000           Total  6381  6311   6269   
5915  27  Chocó  27001  Medio Atrato  2000           Total   141   144    123   
5975  27  Chocó  27001     Río Quito  2000           Total   105    90     90   

      15_20  ...  65_70  70_75  75_80  80_85  85_90  90_95  95_100  \
5745   5498  ...    784    577    386    244    118     51      29   
5915     93  ...     15     13      5      2      0      0       0   
5975     77  ...      9     15      4      0      0      0       0   

      Mujeres_100    yes  oldcode  
5745           16  False    27001  
5915            0   True    27425  
5975            0   True    27600  

[3 rows x 29 columns]

# Sumamos las poblaciones de los municipios que componen cada una de las nuevas regiones conformadas.

In [14]:
for ids in df[df["yes"]].DPMP.unique():
    for year in range(1995,2019):
        for i in range(20):
            df.loc[(df["DPMP"] == str(ids)) & (df["AÑO"] == year) & (~df["yes"]), str(5*i)+"_"+str(5*(i+1))]=df.loc[(df["DPMP"] == str(ids)) & (df["AÑO"] == year) & (~df["yes"]), str(5*i)+"_"+str(5*(i+1))]+sum(df.loc[(df["DPMP"] == str(ids)) & (df["AÑO"] == year) & (df["yes"]), str(5*i)+"_"+str(5*(i+1))])



In [15]:
df.columns

Index(['DP', 'DPNOM', 'DPMP', 'MPIO', 'AÑO', 'ÁREA GEOGRÁFICA', '0_5', '5_10',
       '10_15', '15_20', '20_25', '25_30', '30_35', '35_40', '40_45', '45_50',
       '50_55', '55_60', '60_65', '65_70', '70_75', '75_80', '80_85', '85_90',
       '90_95', '95_100', 'Mujeres_100', 'yes', 'oldcode'],
      dtype='object')

In [16]:
df[(df.DPMP=="27001") & (df["AÑO"]==2000)]

DP  DPNOM   DPMP          MPIO   AÑO ÁREA GEOGRÁFICA   0_5  5_10  10_15  \
5745  27  Chocó  27001        Quibdó  2000           Total  6627  6545   6482   
5915  27  Chocó  27001  Medio Atrato  2000           Total   141   144    123   
5975  27  Chocó  27001     Río Quito  2000           Total   105    90     90   

      15_20  ...  65_70  70_75  75_80  80_85  85_90  90_95  95_100  \
5745   5668  ...    808    605    395    246    118     51      29   
5915     93  ...     15     13      5      2      0      0       0   
5975     77  ...      9     15      4      0      0      0       0   

      Mujeres_100    yes  oldcode  
5745           16  False    27001  
5915            0   True    27425  
5975            0   True    27600  

[3 rows x 29 columns]

# Después de sumar, conservamos solo los municipios originales

In [17]:
df=df[~df['yes']].copy()

In [18]:
del df['yes']
#del df["newcode"]

In [19]:
df["Total"]=sum(df[str(5*i)+"_"+str(5*(i+1))] for i in range(20))+df["Mujeres_100"]


In [20]:
df_try=pd.DataFrame(pd.pivot_table(df[(df["AÑO"]>=1998) & (df["AÑO"]<=2018)],values='Total',index=["DPMP"], columns=["AÑO"], aggfunc="sum").fillna(0).reset_index())
df_try.to_csv("pob_recodificada.csv")
df_try

AÑO    DPMP    1998     1999     2000     2001     2002     2003     2004  \
0     05001  991843  1008226  1024009  1039108  1053805  1068062  1081808   
1     05002   12232    12118    12013    11927    11790    11674    11563   
2     05004    1276     1276     1317     1302     1322     1305     1330   
3     05021    2395     2433     2386     2407     2406     2397     2397   
4     05030   12953    13079    13214    13323    13421    13506    13603   
...     ...     ...      ...      ...      ...      ...      ...      ...   
1059  97666     395      394      444      484      528      539      570   
1060  99001    7657     7800     7916     8040     8154     8299     8395   
1061  99524    3419     3484     3572     3631     3741     3841     3930   
1062  99624    1244     1285     1270     1327     1378     1412     1472   
1063  99773   15455    16201    17023    17819    18604    19399    20240   

AÑO      2005     2006  ...     2009     2010     2011     2012     2013  \
0     1094475  1108756  ...  1150885  1164975  1178109  1190417  1202762   
1       11412    11306  ...    10967    10833    10667    10416    10223   
2        1327     1289  ...     1287     1279     1275     1258     1248   
3        2388     2393  ...     2379     2370     2359     2336     2338   
4       13682    13765  ...    14063    14180    14328    14433    14625   
...       ...      ...  ...      ...      ...      ...      ...      ...   
1059      535      557  ...      678      718      786      852      912   
1060     8646     8759  ...     9098     9187     9300     9381     9453   
1061     3893     3974  ...     4203     4262     4339     4394     4473   
1062     1486     1567  ...     1660     1707     1731     1755     1782   
1063    21111    21968  ...    24730    25683    26601    27559    28487   

AÑO      2014     2015     2016     2017     2018  
0     1216048  1229987  1246042  1263834  1286471  
1       10096     9980     9904     9865     9833  
2        1222     1208     1217     1231     1243  
3        2352     2354     2300     2304     2337  
4       14687    14819    14944    15079    15365  
...       ...      ...      ...      ...      ...  
1059      942      981     1006     1047     1066  
1060     9522     9595     9687     9748     9912  
1061     4517     4572     4616     4717     4724  
1062     1826     1862     1881     1926     1929  
1063    29428    30384    31442    32530    34001  

[1064 rows x 22 columns]

## Limiting the dataframe to contain information only for the time period being considered in this research

In [21]:
df=df[df["AÑO"].isin([ano for ano in range(1998,2019)])].copy()

## Changing the name of the columns to correspond to those of the Cancer datasets (age group)

In [22]:
df.columns

Index(['DP', 'DPNOM', 'DPMP', 'MPIO', 'AÑO', 'ÁREA GEOGRÁFICA', '0_5', '5_10',
       '10_15', '15_20', '20_25', '25_30', '30_35', '35_40', '40_45', '45_50',
       '50_55', '55_60', '60_65', '65_70', '70_75', '75_80', '80_85', '85_90',
       '90_95', '95_100', 'Mujeres_100', 'oldcode', 'Total'],
      dtype='object')

In [23]:
df.columns=['DP', 'DPNOM', 'DPMP', 'MPIO', 'AÑO', 'ÁREA GEOGRÁFICA', '0_5', '5_10',
       '10', '11', '12', '13', '14', '15', '16', '17',
       '18', '19', '20', '21', '22', '23', '24', '25',
       '26', '27', '28','oldcode','Total']

# All information for age groups over 25 are associated (summed up) to the latter.

In [24]:
df["25"]=sum([df["25"],df["26"],df["27"],df["28"]])

# Keeping information for the age groups of interest

In [25]:
df=df[['DP', 'DPNOM', 'DPMP', 'MPIO', 'AÑO', 'ÁREA GEOGRÁFICA', 
       '10', '11', '12', '13', '14', '15', '16', '17',
       '18', '19', '20', '21', '22', '23', '24', '25','Total']].copy()
df

DP      DPNOM   DPMP            MPIO   AÑO ÁREA GEOGRÁFICA     10  \
3       5  Antioquia  05001        Medellín  1998           Total  86309   
4       5  Antioquia  05001        Medellín  1999           Total  87795   
5       5  Antioquia  05001        Medellín  2000           Total  89643   
6       5  Antioquia  05001        Medellín  2001           Total  91471   
7       5  Antioquia  05001        Medellín  2002           Total  93120   
...    ..        ...    ...             ...   ...             ...    ...   
10035  97     Vaupés  97666         Taraira  2018           Total    161   
10062  99    Vichada  99001  Puerto Carreño  2018           Total    977   
10071  99    Vichada  99524    La Primavera  2018           Total    585   
10080  99    Vichada  99624   Santa Rosalía  2018           Total    253   
10089  99    Vichada  99773        Cumaribo  2018           Total   4712   

          11     12     13  ...     17     18     19     20     21     22  \
3      87523  82886  85891  ...  56287  45277  34023  26166  21949  17021   
4      89112  84462  84151  ...  58570  47328  35824  26943  22456  17727   
5      90256  86793  82865  ...  60913  49388  37728  27968  22744  18119   
6      90985  89390  82136  ...  63397  51486  39748  29282  22993  18381   
7      91592  91773  82112  ...  65961  53683  41834  30743  23518  18858   
...      ...    ...    ...  ...    ...    ...    ...    ...    ...    ...   
10035    140     78     62  ...     38     25     16     15      8      5   
10062    950    922    889  ...    466    402    337    275    178     79   
10071    483    401    392  ...    234    198    142    108     87     49   
10080    200    162    142  ...     88     77     64     53     39     20   
10089   4016   3063   2416  ...   1215   1004    795    528    385    233   

          23    24    25    Total  
3      12315  7430  6393   991843  
4      12577  7974  6584  1008226  
5      12915  8382  6784  1024009  
6      13284  8820  7044  1039108  
7      13799  9254  7355  1053805  
...      ...   ...   ...      ...  
10035      6     3     7     1066  
10062     47    26    30     9912  
10071     27    17    27     4724  
10080     13     8     7     1929  
10089    139    81    71    34001  

[22344 rows x 23 columns]

# Creating an auxiliar ID that will be later used for joining the population data set to the Cancer data set

In [26]:
df["id"]=df.apply(lambda x: str(x.AÑO)+"-"+str(x.DPMP),axis=1)

In [27]:
df

DP      DPNOM   DPMP            MPIO   AÑO ÁREA GEOGRÁFICA     10  \
3       5  Antioquia  05001        Medellín  1998           Total  86309   
4       5  Antioquia  05001        Medellín  1999           Total  87795   
5       5  Antioquia  05001        Medellín  2000           Total  89643   
6       5  Antioquia  05001        Medellín  2001           Total  91471   
7       5  Antioquia  05001        Medellín  2002           Total  93120   
...    ..        ...    ...             ...   ...             ...    ...   
10035  97     Vaupés  97666         Taraira  2018           Total    161   
10062  99    Vichada  99001  Puerto Carreño  2018           Total    977   
10071  99    Vichada  99524    La Primavera  2018           Total    585   
10080  99    Vichada  99624   Santa Rosalía  2018           Total    253   
10089  99    Vichada  99773        Cumaribo  2018           Total   4712   

          11     12     13  ...     18     19     20     21     22     23  \
3      87523  82886  85891  ...  45277  34023  26166  21949  17021  12315   
4      89112  84462  84151  ...  47328  35824  26943  22456  17727  12577   
5      90256  86793  82865  ...  49388  37728  27968  22744  18119  12915   
6      90985  89390  82136  ...  51486  39748  29282  22993  18381  13284   
7      91592  91773  82112  ...  53683  41834  30743  23518  18858  13799   
...      ...    ...    ...  ...    ...    ...    ...    ...    ...    ...   
10035    140     78     62  ...     25     16     15      8      5      6   
10062    950    922    889  ...    402    337    275    178     79     47   
10071    483    401    392  ...    198    142    108     87     49     27   
10080    200    162    142  ...     77     64     53     39     20     13   
10089   4016   3063   2416  ...   1004    795    528    385    233    139   

         24    25    Total          id  
3      7430  6393   991843  1998-05001  
4      7974  6584  1008226  1999-05001  
5      8382  6784  1024009  2000-05001  
6      8820  7044  1039108  2001-05001  
7      9254  7355  1053805  2002-05001  
...     ...   ...      ...         ...  
10035     3     7     1066  2018-97666  
10062    26    30     9912  2018-99001  
10071    17    27     4724  2018-99524  
10080     8     7     1929  2018-99624  
10089    81    71    34001  2018-99773  

[22344 rows x 24 columns]

# Reading the Cancer data set and Joining Tables

In [28]:
df_mort=pd.read_csv("output_files/Cancer_1998_2018.csv")


In [29]:
df_mort["codmunicipio"]=df_mort.apply(lambda x: str(x.codmunre).zfill(5),axis=1)
df_mort=df_mort.astype({'ano': 'int32',"area_res":'float32'})
df_mort=pd.DataFrame(pd.pivot_table(df_mort,values='gru_ed2',index=["codmunicipio",'ano'], columns=["gru_ed1"], aggfunc="count").fillna(0).reset_index())



In [30]:
df_mort["mort"]=sum(df_mort[i] for i in range(10,27))


In [31]:
#del df_mort[26]
df_mort.mort.sum()

46752.0

# Se observan 132 casos de fallecidos para los cuales no se conocer la edad, incorrectamente los estabamos sumando antes al grupo de mayor edad.

df_mort=df_mort[~df_mort.codmunicipio.isin(excluded)].copy()

In [32]:
df_mort.columns

Index(['codmunicipio',          'ano',             10,             11,
                   12,             13,             14,             15,
                   16,             17,             18,             19,
                   20,             21,             22,             23,
                   24,             25,             26,         'mort'],
      dtype='object', name='gru_ed1')

# Creating an auxiliar ID to join DataFrames

In [33]:
df_mort["id"]=df_mort.apply(lambda x: str(int(x.ano))+"-"+str(x.codmunicipio),axis=1)


In [34]:
df_both=df.merge(df_mort,how="left",left_on=["DPMP",'AÑO'],right_on=["codmunicipio","ano"])
df_both.to_excel("output_files/revisame.xlsx")


In [35]:
df_both.mort.sum()

46752.0

## Computing the mortality rates for each group

In [36]:
for i in range(10,26):
    df_both["tasa_"+str(i)]=df_both[i]/df_both[f'{i}']*100000


In [37]:
df_both.sort_values(by=["DPMP","AÑO"]).head(50)

DP      DPNOM   DPMP       MPIO   AÑO ÁREA GEOGRÁFICA     10      11  \
0       5  Antioquia  05001   Medellín  1998           Total  86309   87523   
1       5  Antioquia  05001   Medellín  1999           Total  87795   89112   
2       5  Antioquia  05001   Medellín  2000           Total  89643   90256   
3       5  Antioquia  05001   Medellín  2001           Total  91471   90985   
4       5  Antioquia  05001   Medellín  2002           Total  93120   91592   
5       5  Antioquia  05001   Medellín  2003           Total  94451   92407   
6       5  Antioquia  05001   Medellín  2004           Total  95450   93538   
7448    5  Antioquia  05001   Medellín  2005           Total  92223   93032   
7449    5  Antioquia  05001   Medellín  2006           Total  92454   94632   
7450    5  Antioquia  05001   Medellín  2007           Total  92327   96208   
7451    5  Antioquia  05001   Medellín  2008           Total  91873   97578   
7452    5  Antioquia  05001   Medellín  2009           Total  91145   98740   
7453    5  Antioquia  05001   Medellín  2010           Total  90148   99627   
7454    5  Antioquia  05001   Medellín  2011           Total  88995  100286   
7455    5  Antioquia  05001   Medellín  2012           Total  87777  100688   
7456    5  Antioquia  05001   Medellín  2013           Total  86407  100645   
7457    5  Antioquia  05001   Medellín  2014           Total  84901  100095   
7458    5  Antioquia  05001   Medellín  2015           Total  83421   99071   
7459    5  Antioquia  05001   Medellín  2016           Total  82185   97859   
7460    5  Antioquia  05001   Medellín  2017           Total  81145   96580   
21280   5  Antioquia  05001   Medellín  2018           Total  80380   95337   
7       5  Antioquia  05002  Abejorral  1998           Total   1292    1159   
8       5  Antioquia  05002  Abejorral  1999           Total   1246    1106   
9       5  Antioquia  05002  Abejorral  2000           Total   1209    1044   
10      5  Antioquia  05002  Abejorral  2001           Total   1191    1009   
11      5  Antioquia  05002  Abejorral  2002           Total   1180     988   
12      5  Antioquia  05002  Abejorral  2003           Total   1163     970   
13      5  Antioquia  05002  Abejorral  2004           Total   1153     959   
7461    5  Antioquia  05002  Abejorral  2005           Total   1147     952   
7462    5  Antioquia  05002  Abejorral  2006           Total   1125     949   
7463    5  Antioquia  05002  Abejorral  2007           Total   1102     946   
7464    5  Antioquia  05002  Abejorral  2008           Total   1071     939   
7465    5  Antioquia  05002  Abejorral  2009           Total   1044     932   
7466    5  Antioquia  05002  Abejorral  2010           Total   1016     927   
7467    5  Antioquia  05002  Abejorral  2011           Total    988     922   
7468    5  Antioquia  05002  Abejorral  2012           Total    960     912   
7469    5  Antioquia  05002  Abejorral  2013           Total    931     900   
7470    5  Antioquia  05002  Abejorral  2014           Total    892     872   
7471    5  Antioquia  05002  Abejorral  2015           Total    856     848   
7472    5  Antioquia  05002  Abejorral  2016           Total    831     820   
7473    5  Antioquia  05002  Abejorral  2017           Total    803     793   
21281   5  Antioquia  05002  Abejorral  2018           Total    782     781   
14      5  Antioquia  05004   Abriaquí  1998           Total    136     129   
15      5  Antioquia  05004   Abriaquí  1999           Total    140     122   
16      5  Antioquia  05004   Abriaquí  2000           Total    140     120   
17      5  Antioquia  05004   Abriaquí  2001           Total    140     113   
18      5  Antioquia  05004   Abriaquí  2002           Total    142     115   
19      5  Antioquia  05004   Abriaquí  2003           Total    139     107   
20      5  Antioquia  05004   Abriaquí  2004           Total    136     108   
7474    5  Antioquia  05004   Abriaquí  2005           To

In [38]:
df_both=df_both.fillna(0)
df_both.to_csv("output_files/TasasxGrupoEdad.csv")

## Creating a dictionary based on the weights corresponding to each age group according to the WHO (2000 - 2025) Note: Does not sum 100 because it was not necessary to keep the age groups below 10, provided there were no information about breast cancer in the studied period

df_both=pd.read_csv("output_files/TasasxGrupoEdad.csv")

In [39]:
diccionario={10: 8.60,
11: 8.47,
12: 8.22,
13: 7.93,
14: 7.61,
15: 7.15,
16: 6.59,
17 :6.04,
18 :5.37,
19 :4.55,
20 :3.72,
21 :2.96,
22 :2.21,
23 :1.52,
24 :0.91,
25 :0.44+0.15+0.04+0.005}

In [40]:
df_both=df_both.fillna(0)
df_both

DP      DPNOM   DPMP            MPIO   AÑO ÁREA GEOGRÁFICA     10  \
0       5  Antioquia  05001        Medellín  1998           Total  86309   
1       5  Antioquia  05001        Medellín  1999           Total  87795   
2       5  Antioquia  05001        Medellín  2000           Total  89643   
3       5  Antioquia  05001        Medellín  2001           Total  91471   
4       5  Antioquia  05001        Medellín  2002           Total  93120   
...    ..        ...    ...             ...   ...             ...    ...   
22339  97     Vaupés  97666         Taraira  2018           Total    161   
22340  99    Vichada  99001  Puerto Carreño  2018           Total    977   
22341  99    Vichada  99524    La Primavera  2018           Total    585   
22342  99    Vichada  99624   Santa Rosalía  2018           Total    253   
22343  99    Vichada  99773        Cumaribo  2018           Total   4712   

          11     12     13  ...    tasa_16    tasa_17    tasa_18    tasa_19  \
0      87523  82886  85891  ...   8.643167  21.319310  37.546657  23.513506   
1      89112  84462  84151  ...  13.898347  35.854533  29.580798  64.202769   
2      90256  86793  82865  ...  26.862584  32.833714  40.495667  58.312129   
3      90985  89390  82136  ...  14.305221  25.237787  23.307307  47.801147   
4      91592  91773  82112  ...  10.101265  22.740710  27.941807  40.636803   
...      ...    ...    ...  ...        ...        ...        ...        ...   
22339    140     78     62  ...   0.000000   0.000000   0.000000   0.000000   
22340    950    922    889  ...   0.000000   0.000000   0.000000   0.000000   
22341    483    401    392  ...   0.000000   0.000000   0.000000   0.000000   
22342    200    162    142  ...   0.000000   0.000000   0.000000   0.000000   
22343   4016   3063   2416  ...   0.000000   0.000000   0.000000   0.000000   

         tasa_20    tasa_21    tasa_22     tasa_23     tasa_24     tasa_25  
0      64.969808  63.784227  64.626050  105.562322   80.753701   62.568434  
1      29.692313  62.344140  78.975574   95.412260  112.866817   91.130012  
2      57.208238  35.174112  55.190684   61.943477  131.233596  147.405660  
3      23.905471  52.189797  43.523203  105.389943   56.689342   85.178876  
4      68.308233  46.772685  95.450207   94.209725  129.673655  163.154317  
...          ...        ...        ...         ...         ...         ...  
22339   0.000000   0.000000   0.000000    0.000000    0.000000    0.000000  
22340   0.000000   0.000000   0.000000    0.000000    0.000000    0.000000  
22341   0.000000   0.000000   0.000000    0.000000    0.000000    0.000000  
22342   0.000000   0.000000   0.000000    0.000000    0.000000    0.000000  
22343   0.000000   0.000000   0.000000    0.000000    0.000000    0.000000  

[22344 rows x 61 columns]

### Adjusting mortality rate by age (direct method)

In [41]:
df_both["TAE"]=sum(df_both["tasa_"+str(i)]*diccionario[i]/100 for i in range(10,26))


In [42]:
df_both.columns

Index([             'DP',           'DPNOM',            'DPMP',
                  'MPIO',             'AÑO', 'ÁREA GEOGRÁFICA',
                    '10',              '11',              '12',
                    '13',              '14',              '15',
                    '16',              '17',              '18',
                    '19',              '20',              '21',
                    '22',              '23',              '24',
                    '25',           'Total',            'id_x',
          'codmunicipio',             'ano',                10,
                      11,                12,                13,
                      14,                15,                16,
                      17,                18,                19,
                      20,                21,                22,
                      23,                24,                25,
                      26,            'mort',            'id_y',
               'tasa_10',         'tasa_

df_out=df_both[df_both["AÑO"]==2018][["DPMP","TEA"]].copy()
df_out.DPMP=df_out.DPMP.apply(lambda x: str(x).zfill(5) )
df_out.to_csv("TEA_2018.csv")
df_out


# Computando la tasa ajustada de mortalidad por quinquenio

Agregando una variable llamada periodo de acuerdo al año evaluado.

In [43]:
df_both["periodo"]=pd.cut(df_both.AÑO, bins=[0, 2002, 2007, 2012, 100000], labels='1998-2002 2003-2007 2008-2012 2013-2018'.split())
df_both=df_both[['DP', 'DPNOM', 'DPMP', 'MPIO', 'AÑO', 'ÁREA GEOGRÁFICA','periodo','TAE',"mort"]].copy()
df_both.DPMP=df_both.DPMP.apply(lambda x: str(int(x)).zfill(5))

df_out# Se genera un nuevo dataframe que asocia municipios, años y tasas de mortalidad para posteriormente unirlo con las variables socioeconómicas.

In [44]:
df_both.mort.sum()

46752.0

In [45]:
df

DP      DPNOM   DPMP            MPIO   AÑO ÁREA GEOGRÁFICA     10  \
3       5  Antioquia  05001        Medellín  1998           Total  86309   
4       5  Antioquia  05001        Medellín  1999           Total  87795   
5       5  Antioquia  05001        Medellín  2000           Total  89643   
6       5  Antioquia  05001        Medellín  2001           Total  91471   
7       5  Antioquia  05001        Medellín  2002           Total  93120   
...    ..        ...    ...             ...   ...             ...    ...   
10035  97     Vaupés  97666         Taraira  2018           Total    161   
10062  99    Vichada  99001  Puerto Carreño  2018           Total    977   
10071  99    Vichada  99524    La Primavera  2018           Total    585   
10080  99    Vichada  99624   Santa Rosalía  2018           Total    253   
10089  99    Vichada  99773        Cumaribo  2018           Total   4712   

          11     12     13  ...     18     19     20     21     22     23  \
3      87523  82886  85891  ...  45277  34023  26166  21949  17021  12315   
4      89112  84462  84151  ...  47328  35824  26943  22456  17727  12577   
5      90256  86793  82865  ...  49388  37728  27968  22744  18119  12915   
6      90985  89390  82136  ...  51486  39748  29282  22993  18381  13284   
7      91592  91773  82112  ...  53683  41834  30743  23518  18858  13799   
...      ...    ...    ...  ...    ...    ...    ...    ...    ...    ...   
10035    140     78     62  ...     25     16     15      8      5      6   
10062    950    922    889  ...    402    337    275    178     79     47   
10071    483    401    392  ...    198    142    108     87     49     27   
10080    200    162    142  ...     77     64     53     39     20     13   
10089   4016   3063   2416  ...   1004    795    528    385    233    139   

         24    25    Total          id  
3      7430  6393   991843  1998-05001  
4      7974  6584  1008226  1999-05001  
5      8382  6784  1024009  2000-05001  
6      8820  7044  1039108  2001-05001  
7      9254  7355  1053805  2002-05001  
...     ...   ...      ...         ...  
10035     3     7     1066  2018-97666  
10062    26    30     9912  2018-99001  
10071    17    27     4724  2018-99524  
10080     8     7     1929  2018-99624  
10089    81    71    34001  2018-99773  

[22344 rows x 24 columns]

In [46]:
df_mort.ano.unique()

array([1998, 1999, 2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008,
       2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018])

In [47]:
df_out=pd.DataFrame(df.DPMP.unique())
df_out.columns=["DPMP"]

for y in list(df_both[df_both["AÑO"]>0]["AÑO"].unique()):
    df_out["mort_"+str(y)]=df_out.DPMP.map(df_both[df_both["AÑO"]==y].set_index("DPMP")["mort"])    
    
for y in list(df_both[df_both["AÑO"]>0]["AÑO"].unique()):
    df_out["TAE_"+str(y)]=df_out.DPMP.map(df_both[df_both["AÑO"]==y].set_index("DPMP")["TAE"])



df_out.DPMP=df_out.DPMP.apply(lambda x: str(int(x)).zfill(5))
df_out=df_out.fillna(0)

In [48]:
dictio=pd.DataFrame(pd.read_csv("data/DIVIPOLA_Municipios.csv"))
dictio.COD=dictio.COD.astype(str)
dictio.COD=dictio.COD.apply(lambda x: x.zfill(5))
print(dictio.columns)
dictio=dictio.set_index('COD').T.to_dict()
for i in dictio.keys():
    dictio[i]=dictio[i]['MUNICIPIO']

df_out["MPIO"]=df_out.DPMP.map(dictio).apply(lambda x: str(x).capitalize())

Index(['COD', 'MUNICIPIO'], dtype='object')


In [49]:
df_out.to_csv("output_files/TAE_&_mortality.csv")

In [50]:
df_out.to_excel("output_files/TAE_&_mortality.xlsx")

In [51]:
df_out

DPMP  mort_1998  mort_1999  mort_2000  mort_2001  mort_2002  mort_2003  \
0     05001      117.0      140.0      152.0      121.0      156.0      181.0   
1     05002        0.0        0.0        1.0        0.0        0.0        0.0   
2     05004        0.0        0.0        0.0        0.0        0.0        0.0   
3     05021        1.0        0.0        0.0        0.0        0.0        0.0   
4     05030        0.0        1.0        2.0        0.0        0.0        1.0   
...     ...        ...        ...        ...        ...        ...        ...   
1059  97666        0.0        0.0        0.0        0.0        0.0        0.0   
1060  99001        0.0        2.0        1.0        0.0        0.0        0.0   
1061  99524        0.0        0.0        0.0        0.0        0.0        0.0   
1062  99624        0.0        0.0        0.0        0.0        0.0        0.0   
1063  99773        0.0        0.0        0.0        0.0        0.0        0.0   

      mort_2004  mort_2005  mort_2006  ...   TAE_2010   TAE_2011   TAE_2012  \
0         166.0      138.0      181.0  ...  15.113675  15.304540  15.472662   
1           1.0        2.0        0.0  ...   8.551724   0.000000   0.000000   
2           0.0        0.0        0.0  ...   0.000000   0.000000   0.000000   
3           0.0        0.0        0.0  ...   0.000000   0.000000   0.000000   
4           0.0        1.0        1.0  ...  19.424761  24.301651   5.994730   
...         ...        ...        ...  ...        ...        ...        ...   
1059        0.0        0.0        0.0  ...   0.000000   0.000000   0.000000   
1060        1.0        1.0        0.0  ...  41.081081   0.000000   0.000000   
1061        0.0        0.0        0.0  ...   0.000000   0.000000   0.000000   
1062        0.0        0.0        0.0  ...   0.000000  70.860215   0.000000   
1063        0.0        0.0        0.0  ...   0.000000   0.000000   0.000000   

       TAE_2013    TAE_2014   TAE_2015   TAE_2016   TAE_2017   TAE_2018  \
0     15.775365   15.975835  14.570198  16.403328  16.935918  14.176541   
1     18.048366    7.322835   9.263804  28.786943   7.106109   0.000000   
2      0.000000  139.085297  69.062500   0.000000  82.222222   0.000000   
3      0.000000    0.000000   0.000000   0.000000   0.000000   0.000000   
4      6.757991   13.481400   5.118110  18.817483  17.079392   4.846570   
...         ...         ...        ...        ...        ...        ...   
1059   0.000000    0.000000   0.000000   0.000000   0.000000   0.000000   
1060   0.000000    0.000000   0.000000  14.583333  16.723164   0.000000   
1061   0.000000    0.000000   0.000000   0.000000   0.000000   0.000000   
1062   0.000000    0.000000   0.000000   0.000000   0.000000   0.000000   
1063   0.000000    0.000000   0.000000   5.354610   6.034483   0.000000   

                MPIO  
0           Medellín  
1          Abejorral  
2           Abriaquí  
3         Alejandría  
4              Amagá  
...              ...  
1059         Taraira  
1060  Puerto carreño  
1061    La primavera  
1062   Santa rosalía  
1063        Cumaribo  

[1064 rows x 44 columns]